src/unix/linux-core.c

```diff
@@ -983,3 +983,49 @@ void uv__set_process_title(const char* title) {
   prctl(PR_SET_NAME, title);  /* Only copies first 16 characters. */
 #endif
 }
+
+int uv__tcp_enable_largesocket(uv_tcp_t *tcp) {
+  int rmem;
+  int wmem;
+  int r;
+  FILE* fp = uv__open_file("/proc/sys/net/core/rmem_max");
+  if (fp == NULL)
+    return UV_ENOSYS;
+
+  rmem = 0;
+  wmem = 0;
+  if (fscanf(fp, "%d", &rmem) != 1) {
+    fclose(fp);
+    return UV_ENOSYS;
+  }
+
+  fclose(fp);
+
+  if (rmem <= STDTCPWINDOW)
+    return UV_ENOSYS;
+
+  fp = uv__open_file("/proc/sys/net/core/wmem_max");
```

Comment:
this repeated chunk of code could be refactored into a small file-static helper function